frontend/src/metabase/lib/dom.js

```diff
@@ -220,7 +220,25 @@ export function constrainToScreen(element, direction, padding) {
   return false;
 }
 
-const isAbsoluteUrl = url => url.startsWith("/");
+function getSitePath() {
+  return new URL(MetabaseSettings.get("site-url")).pathname.toLowerCase();
+}
+
+function isMetabaseUrl(url) {
```

Comment:
@WiNloSt previously we handled invalid URLs in `getOrigin`, now functions will throw. Is it expected?

Reply:
I can add a test for this, but seems like `getOrigin` [was introduced without `base`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) (the 2nd) option when constructing the URL, that's when it can throw error, but it seems impossible to fail to create a URL if you give a correct `base` (which we do `window.location.origin`).

Reply:
Added tests in https://github.com/metabase/metabase/pull/35975/commits/2a77c3041174167098a9f0e3eeebc2702e634789